src/finetuning_scheduler/fts.py

```diff
@@ -671,12 +672,14 @@ def on_train_epoch_start(self, trainer: "pl.Trainer", pl_module: "pl.LightningMo
             pl_module (:external+pl:class:`~pytorch_lightning.core.module.LightningModule`): The
                 :external+pl:class:`~pytorch_lightning.core.module.LightningModule` object
         """
+        if self.first_epoch_number is None:
+            self.first_epoch_number = trainer.current_epoch
```

Comment:
I think the best place to store this state is within ``_fts_state`` so I've moved it there. Rather than setting this ``on_train_epoch_start``, I think it's better to set it ``on_fit_start`` if it's not already set so have done so. I think it's a good idea to avoid assuming the fine-tuning training session would start on epoch 0, though I haven't fully tested supporting starting FTS-enabled training sessions from non-FTS checkpoints (I'll have to think through the challenges, there could be some limitations there). I'm interested, are there particular use cases you have in mind for starting FTS training sessions after a non-zero number or epochs have been run? Thanks again for the contribution @solalatus!

Reply:
First of all thanks for taking the time for this PR, and the library in general! Great stuff!
The situation here was particular. I have trained a model inn Darts, saved it, loaded it and wanted to finetune. Now as it is implemented in Darts, it continued the counting of epochs (as well as restoring optimizer state, etc.) so I bumped into this problem, I have tried to help in a PR over at Darts also to maybe set this strait, but is a bit more effort on that side (kudos to the people there pushing it). So I used this hack to get through the problem. Albeit I thought it is general enough as a problem to be looked at.